Set number of args to 0 for no-replace and list

dongsupark opened the pull request:
In case of clap 4, it is necessary to set the number of args to 0, if the option does not take any parameter.

dongsupark committed:
Set number of args to 0 for no-replace and list

In case of clap 4, it is necessary to set the number of args to 0,
if the option does not take any parameter.

dongsupark commented:
CI passed https://github.com/flatcar/scripts/pull/1393#issuecomment-1812450120

pothos commented:
Better now but why does it always print the full list even if `-l` is not specified? Maybe a wrong default?

dongsupark committed:
print out list only in case given by user

List of the current keys should be printed only when the user specified
the option --list. If not, simply only sync.

dongsupark commented:
Fixed the issue of printing list by default.

dongsupark commented:
CI passed https://github.com/flatcar/scripts/pull/1393
